sources/packages/backend/libs/integrations/src/cas/cas.service.ts

```diff
@@ -129,26 +131,62 @@ export class CASService {
     supplierData: CreateSupplierAndSiteData,
   ): Promise<CreateSupplierAndSiteResponse> {
     const url = `${this.casIntegrationConfig.baseUrl}/cfs/supplier/`;
+    return this.processSupplierAndSite(url, supplierData);
+  }
+
+  /**
+   * Create supplier site for existing supplier.
+   * @param supplierData data to be used for supplier and site creation.
+   * @returns submitted data and CAS response.
+   */
+  async createExistingSupplierAndSite(
+    supplierData: CreateExistingSupplierAndSiteData,
+  ): Promise<CreateSupplierAndSiteResponse> {
+    const url = `${this.casIntegrationConfig.baseUrl}/cfs/supplier/${supplierData.supplierNumber}/site`;
+    return this.processSupplierAndSite(url, supplierData);
+  }
+
+  /**
+   * Process the API call to create supplier and site.
+   * @param url API url to create a new or existing supplier and site.
+   * @param submittedData data submitted to the API.
+   * @returns submitted data and CAS response
+   */
+  async processSupplierAndSite(
+    url: string,
+    supplierData: CreateSupplierAndSiteData | CreateExistingSupplierAndSiteData,
+  ): Promise<CreateSupplierAndSiteResponse> {
     try {
       const config = await this.getAuthConfig();
-      const submittedData: CreateSupplierAndSiteSubmittedData = {
-        SupplierName: formatUserName(
-          supplierData.firstName,
-          supplierData.lastName,
-        ),
-        SubCategory: "Individual",
-        Sin: supplierData.sin,
-        SupplierAddress: [
-          {
-            AddressLine1: formatAddress(supplierData.supplierSite.addressLine1),
-            City: formatCity(supplierData.supplierSite.city),
-            Province: supplierData.supplierSite.provinceCode,
-            Country: "CA",
-            PostalCode: formatPostalCode(supplierData.supplierSite.postalCode),
-            EmailAddress: supplierData.emailAddress,
-          },
-        ],
+      const supplierAddress = {
+        AddressLine1: formatAddress(supplierData.supplierSite.addressLine1),
+        City: formatCity(supplierData.supplierSite.city),
+        Province: supplierData.supplierSite.provinceCode,
+        Country: "CA",
+        PostalCode: formatPostalCode(supplierData.supplierSite.postalCode),
+        EmailAddress: supplierData.emailAddress,
       };
+      let submittedData:
+        | CreateSupplierAndSiteSubmittedData
+        | CreateExistingSupplierAndSiteSubmittedData;
+      if (supplierData instanceof CreateExistingSupplierAndSiteData) {
+        supplierAddress["AddressLine2"] = "";
+        supplierAddress["AddressLine3"] = "";
```

Comment:
Does the create site require `AddressLine2` and `AddressLine3` to be set to empty string?